packages/uploadthing/src/sveltekit.ts

```diff
@@ -0,0 +1,33 @@
+import type { Handle } from "@sveltejs/kit";
+
+import type { Json } from "@uploadthing/shared";
+
+import type { RouterWithConfig } from "./internal/handler";
+import type { FileRouter } from "./internal/types";
+import type { CreateBuilderOptions } from "./internal/upload-builder";
+import { createBuilder } from "./internal/upload-builder";
+import { createNextRouteHandler } from "./next";
+
+export type { FileRouter } from "./internal/types";
+
+export const createUploadthing = <TErrorShape extends Json>(
+  opts?: CreateBuilderOptions<TErrorShape>,
+) => createBuilder<"app", TErrorShape>(opts);
+
+export const createUploadthingHandle = <TRouter extends FileRouter>(
+  opts: RouterWithConfig<TRouter>,
+): Handle => {
+  const { GET, POST } = createNextRouteHandler(opts);
+  return ({ event, resolve }) => {
+    const { request, url } = event;
+    if (url.pathname.startsWith("/api/uploadthing")) {
+      if (request.method === "POST") {
+        return POST(request);
+      }
+      if (request.method === "GET") {
+        return GET();
+      }
+    }
+    return resolve(event);
+  };
+};
```

Comment:
Can't sveltekit just use the /server entrypoint? Trying to avoid too many entrypoints, especially when they're all using Request/Response

Reply:
I made this change for now. If there is a good reason not to, we can revert

Reply:
I do think having the handle hook is useful, but maybe I can export it from the svelte package instead

Reply:
I went ahead and created the handle hook in the svelte package, although I had to expose the `RouterWithConfig` type, let me know if that's ok

Reply:
> I do think having the handle hook is useful

Why? I'm not saying it's bad I'm just trying to understand what the benefit is?

> I went ahead and created the handle hook in the svelte package

That doesnt make much sense to me, the client side libs doesnt need server code. We can expose a svelte-kit entrypoint if there is a reason for it, but to me it looks like svelte is request/response based so I don't see a reason you can't use that?

I'm also very lost on the difference between server hooks and api routes so might be completely off

Reply:
A server hook in sveltekit acts like a middleware, it can be used for stuff like authentication for example. In this case my idea was to use a server hook to handle the uploadthing api routes, instead of creating a whole extra file to create an actual api route. I've seen other sveltekit libraries do a similar thing, like trpc-sveltekit. Maybe calling it useful was not the right word, it's more like a nice thing to have.

We could also just create the api routes file like with other frameworks, if you think the server hook thing is confusing/not worth it.

Reply:
Decided to let go of the handle hook and just use an API route 😅